include/seqan3/argument_parser/validators.hpp

```diff
@@ -178,11 +178,38 @@ class value_list_validator
     //!\brief Type of values that are tested by validator
     using value_type = option_value_type;
 
-    /*!\brief Constructing from a vector.
-     * \param[in] v The vector of valid values to test.
+    /*!\name Constructors, destructor and assignment
+     * \{
      */
-    value_list_validator(std::vector<value_type> v) : values{std::move(v)}
-    {}
+    value_list_validator() = default;                                         //!< Defaulted.
+    value_list_validator(value_list_validator const &) = default;             //!< Defaulted.
+    value_list_validator(value_list_validator &&) = default;                  //!< Defaulted.
+    value_list_validator & operator=(value_list_validator const &) = default; //!< Defaulted.
+    value_list_validator & operator=(value_list_validator &&) = default;      //!< Defaulted.
+    ~value_list_validator() = default;                                        //!< Defaulted.
+
+    /*!\brief Constructing from a range.
+     * \tparam range_type The type of range; must model std::ranges::forward_range.
+     * \param[in] rng The range of valid values to test.
+     */
+    template <std::ranges::forward_range range_type>
+    value_list_validator(range_type && rng) : values{rng.begin(), rng.end()}
```

Comment:
Actually, you are always copying here. So I guess it would be beneficial to take a copy of the range instead and then use `std::ranges::move()` in the constructor body. Then you still make a copy for lvalues but for rvalues you don't. Not sure, but maybe the compiler is smart enough to in-place allocate in the target vector. But that I don't know since it involves memory allocation etc.

Reply:
Does `std::ranges::move` work on each element of the range?